src/main/cli/commandLine.ts

```diff
@@ -0,0 +1,135 @@
+// ==LICENSE-BEGIN==
+// Copyright 2017 European Digital Reading Lab. All rights reserved.
+// Licensed to the Readium Foundation under one or more contributor license agreements.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file exposed on Github (readium) in the project repository.
+// ==LICENSE-END==
+
+import { readerActions } from "readium-desktop/common/redux/actions";
+import { container } from "readium-desktop/main/di";
+import { Store } from "redux";
+import * as yargs from "yargs";
+
+import { OpdsFeedRepository } from "readium-desktop/main/db/repository/opds";
+import { PublicationRepository } from "readium-desktop/main/db/repository/publication";
+import { RootState } from "readium-desktop/main/redux/states";
+import { CatalogService } from "readium-desktop/main/services/catalog";
+import { URL } from "url";
+
+export interface ICliParam {
+    readonly argv: yargs.Arguments;
+    quit: boolean;
+}
+
+export interface ICli {
+    name: string;
+    fct: (o: ICliParam) => Promise<boolean>;
+    help: string[];
+}
+
+export const cli: ICli[] = [
+    {
+        name: "_",
+        fct: async ({ argv }) => {
+            let publicationOpenRequested = false;
+            let returnValue = true;
+
+            for (const filePath of argv._) {
+                // import and read publication
+                const catalogService = container.get("catalog-service") as CatalogService;
+                const publication = await catalogService.importFile(filePath);
+                const store = container.get("store") as Store<RootState>;
+                if (publication) {
+                    if (!publicationOpenRequested) {
+                        store.dispatch({
+                            type: readerActions.ActionType.OpenRequest,
+                            payload: {
+                                publication: {
+                                    identifier: publication.identifier,
+                                },
+                            },
+                        });
+                        publicationOpenRequested = true;
+                    }
+                } else {
+                    process.stderr.write(`Publication error for "${filePath}"\n`);
+                    returnValue = false;
+                }
+            }
+            return returnValue;
+        },
+        help: [],
+    },
+    {
+        name: "import",
+        fct: async ({ argv }) => {
+            const catalogService = container.get("catalog-service") as CatalogService;
+            await catalogService.importFile(argv.import as string);
+            return true;
```

Comment:
For MacOS, when the application is already open, and "open with Thorium" is invoked, a special event is triggered:
https://github.com/electron/electron/blob/master/docs/api/app.md#event-open-file-macos

There is already placeholder code (but no functionality attached to it):
https://github.com/readium/readium-desktop/blob/541e48bbe45407a8918018e0076d4ca5ebecd68b/src/main.ts#L254-L261

For this PR to be complete for MacOS (double-click on EPUB file, or "open with"), I think we should also invoke `catalogService.importFile()` in the event listeners. What do you think?